### DISCUSSION TITLE
Neovide is randomly working on Wayland and arch linux!

### DISCUSSION THREAD

**Discussion by Feel-ix-343** [neovide/neovide#1906] [General]
I'm sure there was some pull request or something, but I don't know which one. However, I recently updated neovide (git version?) and now smooth scrolling is working AND I don't need to launch it on xwayland (or what ever it is called) anymore!

Some body did some amazing work.

**Comment by fredizzimo** [comment]
We merged this https://github.com/neovide/neovide/pull/1789, which in turn was enabled by the hard work of the winit team, by finally merging the new keyboard API https://github.com/rust-windowing/winit/pull/2662. A huge thank to  @maroider, @arturkovacs, @kchibisov and everyone else that has been working on that.

For a long time Neovide was stuck in a limbo, where we were using a custom version of winit that included the work in progress state of the keyboard API. But that also meant that we were mostly unable to merge in new fixes, which includes proper Wayland support. But now that the keyboard API is merged, we are able to track the master, and soon the release version of winit again, with all the current and future support it provides.

So, the future for Neovide looks bright, with for example more keyboard input fixes coming soon https://github.com/neovide/neovide/pull/1899 and also an improved rendering loop, that makes "smooth" scrolling much smoother https://github.com/neovide/neovide/pull/1870.
